src/Sylius/Bundle/ResourceBundle/Controller/StateMachine.php

```diff
@@ -46,12 +48,20 @@ public function can(RequestConfiguration $configuration, ResourceInterface $reso
     /**
      * {@inheritdoc}
      */
-    public function apply(RequestConfiguration $configuration, ResourceInterface $resource): void
+    public function apply(RequestConfiguration $configuration, ResourceInterface $resource): bool
     {
         if (!$configuration->hasStateMachine()) {
             throw new \InvalidArgumentException('State machine must be configured to apply transition, check your routing.');
         }
 
-        $this->stateMachineFactory->get($resource, $configuration->getStateMachineGraph())->apply($configuration->getStateMachineTransition());
+        try {
+            if (!$this->stateMachineFactory->get($resource, $configuration->getStateMachineGraph())->apply($configuration->getStateMachineTransition())) {
```

Comment:
Could it be solved by using `can` method on state machine factory before we call `apply`?